### PR TITLE
Review use of `Enum`s

### PR DESCRIPTION
**Pull request by tcompa**
Close #953 

## Checklist before merging
- [x] I added an appropriate entry to `CHANGELOG.md`

### CODE REVIEW

File: fractal_server/app/models/job.py [hunk 1]
```diff
@@ -115,5 +92,10 @@ class Config:
     end_timestamp: Optional[datetime] = Field(
         default=None, sa_column=Column(DateTime(timezone=True))
     )
-    status: JobStatusType = JobStatusType.SUBMITTED
+    status: str = JobStatusType.SUBMITTED
     log: Optional[str] = None
+
+    @validator("status")
```
Comment (tcompa) [inline fractal_server/app/models/job.py:98]: @mfranzon let's decide whether this validator is actually valuable

**Comment by tcompa** [comment]
I'm pretty convinced that the bulk of the PR is fine, just removing all `Enum`s from `app/models`, and only keeping them within `app/schemas`. I'm left with a doubt about whether to include a validator in `ApplyWorkflow` - cc @mfranzon

File: fractal_server/app/models/job.py (hunk 1 above)
Comment (mfranzon) [reply fractal_server/app/models/job.py:98]: We can relax the validator as long as there is no user interaction (PATCH) with the job_status

**Review by mfranzon — APPROVED** [review]

File: fractal_server/app/models/job.py (hunk 1 above)
Comment (tcompa) [reply fractal_server/app/models/job.py]: Agreed. It doesn't add so much, apart from complexity. Now removed with https://github.com/fractal-analytics-platform/fractal-server/pull/974/commits/83160692fee6a872a411e3c25f0772409ce6ffe8.

**Pull request merged**